### PR TITLE
Playlist loader cleanup

### PR DESCRIPTION
**Pull request by gesinger**
## Description
ES6ified and cleaned up a couple functions and added tests for them.

## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [X] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/liwecukasi/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/playlist-loader.js [hunk 1]
```diff
@@ -52,59 +50,58 @@ const updateSegments = function(original, update, offset) {
...
+    if (segment.key && !segment.key.resolvedUri) {
+      segment.key.resolvedUri = resolveUrl(mergedPlaylist.resolvedUri, segment.key.uri);
+    }
+    if (segment.map && !segment.map.resolvedUri) {
+      segment.map.resolvedUri = resolveUrl(mergedPlaylist.resolvedUri, segment.map.uri);
+    }
+  });
+
+  // TODO Right now in the playlists array there are two references to each playlist, one
+  // that is referenced by index, and one by URI. The index reference may no longer be
```
Comment (mjneil) [inline src/playlist-loader.js]: I'm pretty sure everywhere within contrib-hls uses playlists as an array and would require that the index based referenced be updated in order to get the new segments

Comment (gesinger) [reply src/playlist-loader.js]: It might be worth considering, but can take out the TODO for now so it doesn't produce excess noise in the linter.

File: src/playlist-loader.js [hunk 2]
```diff
@@ -52,59 +50,82 @@ const updateSegments = function(original, update, offset) {
   * master playlist with the updated media playlist merged in, or
   * null if the merge produced no change.
   */
-const updateMaster = function(master, media) {
-  let changed = false;
-  let result = mergeOptions(master, {});
+export const updateMaster = function(master, media) {
```
Comment (mjneil) [inline src/playlist-loader.js]: surprised you didn't use arrow notation here

File: test/playlist-loader.test.js [hunk 3]
```diff
@@ -655,57 +655,50 @@ QUnit.test('resolveMediaGroupUris resolves media group URIs', function(assert) {
...
           }
         },
-        AUDIO2: {
-          LOW: {
-            uri: 'audio-2-low-uri'
-          },
-          HIGH: {
-            uri: 'audio-2-high-uri'
+        high: {
+          main: {},
```
Comment (mjneil) [inline test/playlist-loader.test.js]: not necessary, but if you want to cover another edge case, you can have a muxed audio label without a uri in one audio group, and the same labeled audio in another group can have unmuxed so it would have a uri. Playlist loader doesn't care about this detail though, its taken into account in media-groups, but incase you wanted another test case

File: test/playlist-loader.test.js [hunk 4]
```diff
@@ -28,6 +34,745 @@ QUnit.module('Playlist Loader', {
...
 
+QUnit.test('updateSegments copies over properties', function(assert) {
+  assert.deepEqual(
+    [
+      { uri: 'test-uri-0', startTime: 0, endTime: 10 },
+      {
+        uri: 'test-uri-1',
+        startTime: 10,
+        endTime: 20,
+        attributes: { BANDWIDTH: 99, height: 4 }
```
Comment (mjneil) [inline test/playlist-loader.test.js]: You don't need to change these because whats really being tested is object copy and merging, which doesn't care about the content, but it was a little confusing to see a combination of segment and playlist objects being used to test `updateSegments`.
